ir/expression.def

```diff
@@ -46,7 +46,8 @@ abstract Operation_Binary : Operation {
             type = left->type; }
     toString {
         std::stringstream tmp;
-        tmp << DBPrint::Prec_Low << *this;
+        tmp << DBPrint::Prec_Low;
```

Comment:
we probably want another DBPrint flag here to disable the printing of mangled names/only print origName as we're using this for `toString` which is user facing.  Improvement unrelated to this change really.

Reply:
So, I guess the problem with
```
    toString {
        return left->toString() + " " + getStringOp() + " " + right->toString();
```
is operator precedence?

Reply:
Yes -- dbprint automatically inserts parens as needed for precedence.

Reply:
A DBPrint flag may be the way to go as that would allow us to unify all this dbprint/toString stuff with the selection of user facing vs developer facing marked by the flag.  error_helper could then just set the flag.

Reply:
Yeah. testdata/p4_16_samples_outputs/array_field.p4-stderr is a great example where we produce something internal in the warnings which might be confusing to users

Reply:
Moved this to https://github.com/p4lang/p4c/issues/4698